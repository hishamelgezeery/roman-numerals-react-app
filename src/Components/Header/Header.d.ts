import React, { Component, CSSProperties } from 'react'

type bgTypes = 'primary' |'secondary' | 'success' | 'danger' |'warning' |'info' | 'light' | 'dark' | 'muted' | 'white'

type variantTypes = 'light' |'dark'

type expandTypes = 'sm' | 'md' | 'lg' | 'xl'

export interface HeaderProps {
 /**
 * The background color of the Navbar
 */
 backgroundColor: bgTypes;
 /**
 * The variant color scheme of the Navbar
 */
 variantType: variantTypes;
 /**
 * The expand size of the Navbar
 * @default lg
 */
 expandType: expandTypes;
 /**
 * The main title of the navbar
 */
 navbarText: String;
 /**
 * The url to navigate to on main navbar title press
 */
 navbarLink: String;
 /**
 * The object to provide CSS properties to the Navbar
 */
 style: CSSProperties;
}

export default class Header extends Component<HeaderProps> { }
