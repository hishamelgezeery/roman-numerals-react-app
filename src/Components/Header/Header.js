import React from "react";
import PropTypes from "prop-types";
import { Navbar } from "react-bootstrap";

const Header = ({
  navbarLink,
  backgroundColor,
  navbarText,
  variantType,
  expandType,
  style,
}) => {
  return (
    <Navbar
      bg={backgroundColor}
      variant={variantType}
      expand={expandType}
      style={style}
    >
      <Navbar.Brand href={navbarLink}>{navbarText}</Navbar.Brand>
    </Navbar>
  );
};

Header.propTypes = {
  navbarLink: PropTypes.string,
  backgroundColor: PropTypes.oneOf([
    "primary",
    "secondary",
    "success",
    "danger",
    "warning",
    "info",
    "light",
    "dark",
    "muted",
    "white",
  ]),
  variantType: PropTypes.oneOf(["light", "dark"]),
  expandType: PropTypes.oneOf(["sm" | "md" | "lg" | "xl"]),
  navbarText: PropTypes.string,
  style: PropTypes.object,
};

Header.defaultProps = {
  navbarLink: "#",
  navbarText: "",
};

export default Header;
