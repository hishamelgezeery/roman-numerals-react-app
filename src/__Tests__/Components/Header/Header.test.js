import renderer from 'react-test-renderer'
import Header from '../../../Components/Header'

describe('Header Component', () => {
  it('should render correctly', () => {
    const instance = renderer.create(<Header navbarLink={"#"} backgroundColor={'light'} navbarText={'test navbar'} variantType={'dark'}
        />).toJSON()
    expect(instance).toMatchSnapshot()
  })
})
