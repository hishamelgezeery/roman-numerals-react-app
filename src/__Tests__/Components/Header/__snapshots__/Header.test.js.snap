// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-light"
>
  <a
    className="navbar-brand"
    href="#"
  >
    test navbar
  </a>
</nav>
`;
